### PR TITLE
fix: enabling more tests with strict schema validation 

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Run more tests with strict response schema validation and fixing health schema accordingly.

### CODE REVIEW

File: src/test/e2e/api/openapi/__snapshots__/openapi.e2e.test.ts.snap [hunk 1]
```diff
@@ -1323,6 +1327,13 @@ exports[`should serve the OpenAPI spec 1`] = `
             "example": false,
             "type": "boolean",
           },
+          "strategies": {
```
Comment (kwasniew) [inline src/test/e2e/api/openapi/__snapshots__/openapi.e2e.test.ts.snap:1330]: added this as some clients may depend on this legacy structure

**Review by sjaanus — APPROVED** [review]

File: src/test/e2e/api/admin/constraints.e2e.test.ts [hunk 2]
```diff
@@ -9,7 +9,13 @@ const PATH = '/api/admin/constraints/validate';
 
 beforeAll(async () => {
     db = await dbInit('constraints', getLogger);
-    app = await setupApp(db.stores);
+    app = await setupAppWithCustomConfig(db.stores, {
+        experimental: {
+            flags: {
+                strictSchemaValidation: true,
```
Comment (gastonfournier) [inline src/test/e2e/api/admin/constraints.e2e.test.ts:15]: :raised_hands:

File: src/lib/openapi/spec/health-overview-schema.ts [hunk 3]
```diff
@@ -23,6 +23,7 @@ export const healthOverviewSchema = {
         },
         description: {
             type: 'string',
+            nullable: true,
```
Comment (gastonfournier) [inline src/lib/openapi/spec/health-overview-schema.ts:26]: I guess this is different than not being required. Right?

**Review by gastonfournier — APPROVED** [review]
Nice!

File: src/lib/openapi/spec/health-overview-schema.ts (hunk 3 above)
Comment (kwasniew) [reply src/lib/openapi/spec/health-overview-schema.ts:26]: this is when description is there and set to null. We were getting those values in the value space, so adding them to the type space

**Pull request merged**